### PR TITLE
Move sender receiver code to transfer module

### PR DESCRIPTION
**Pull request by berryve**
What's in this PR:
1. Remove listening for account creation in Onboarding.java - (it is not necessary for our sample app and it caused occasional failures).
2. Moved all sender/receiver generic code to separate transfer module so that it can be reused by one wallet and kinit.
3. Make naming of the account info that is being requested more generic (i.e. instead of returning 'public address' we return a generic account data string. (Kinit - in the one wallet module case we will be returning the linking transaction envelope as a string instead of a public address)
4. Some improvements in TransferManager

### CODE REVIEW

**Comment by katzio** [comment]
Added some tests

File: baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/sampleWallet/OnBoarding.java [hunk 1]
```diff
@@ -44,17 +44,18 @@
...
-            listenerRegistration.remove();
-            fireOnFailure(callbacks, new TimeoutException("Waiting for account creation event time out"));
-        };
-        listenerRegistration = account.addAccountCreationListener(data -> {
-            listenerRegistration.remove();
-            handler.removeCallbacks(accountCreationListeningTimeout);
-            fireOnSuccess(callbacks);
-        });
-        handler.postDelayed(accountCreationListeningTimeout, 10 * DateUtils.SECOND_IN_MILLIS);
+ //       Runnable accountCreationListeningTimeout = () -> {
```
Comment (katzio) [inline baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/sampleWallet/OnBoarding.java]: @berryve remove commented section?

File: baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/sampleWallet/OnBoarding.java [hunk 2]
```diff
@@ -77,6 +78,10 @@ public void onResponse(@NonNull Call call, @NonNull Response response) {
                         if (code != 200) {
                             fireOnFailure(callbacks, new Exception("Create account - response code is " + response.code()));
                         }
+                        else {
+                            //handler.removeCallbacks(accountCreationListeningTimeout);
```
Comment (katzio) [inline baseSampleApp/src/main/java/org/kinecosystem/baseSampleApp/sampleWallet/OnBoarding.java]: same here

File: receiverSampleApp/src/main/java/org/kinecosystem/sampleReceiverApp/AccountInfoActivity.java [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-import org.kinecosystem.appsdiscovery.receiver.view.AccountInfoActivityBase;
+import org.kinecosystem.transfer.receiver.view.AccountInfoActivityBase;
 import org.kinecosystem.baseSampleApp.sampleWallet.SampleWallet;
 
 
 public class AccountInfoActivity extends AccountInfoActivityBase {
 
     @Override
-    public String getPublicAddress() {
+    public String getAccountInfo() {
```
Comment (katzio) [inline receiverSampleApp/src/main/java/org/kinecosystem/sampleReceiverApp/AccountInfoActivity.java]: 👏

**Review by katzio — APPROVED** [review]
I Approve but I need someone to take a look at the unittests I added (My PR had conflicts with this one, was easier to apply changes here)

File: appsDiscovery/build.gradle [hunk 4]
```diff
@@ -5,7 +5,7 @@ apply plugin: 'kotlin-android'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-
+    testOptions.unitTests.includeAndroidResources = true
```
Comment (shayb) [inline appsDiscovery/build.gradle:8]: need to change readme pls

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/presenter/AppInfoPresenter.kt [hunk 5]
```diff
@@ -153,7 +153,8 @@ class AppInfoPresenter(private val appName: String?, private val repository: Dis
         repository.clearReceiverAppPublicAddress()
         app?.launchActivity?.let { activityPath ->
             app?.identifier?.let { receiverPkg ->
-                val started = transferManager.startTransferRequestActivity(receiverPkg, activityPath)
+                val started = transferManager.startTransferRequestActivity(REMOTE_PUBLIC_ADDRESS_REQUEST_CODE,
+                        receiverPkg, activityPath)
```
Comment (shayb) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/presenter/AppInfoPresenter.kt]: why need request code

File: receiverSampleApp/src/main/java/org/kinecosystem/sampleReceiverApp/AccountInfoActivity.java (hunk 3 above)
Comment (shayb) [reply receiverSampleApp/src/main/java/org/kinecosystem/sampleReceiverApp/AccountInfoActivity.java]: why do you clap?

File: transfer/src/main/java/org/kinecosystem/transfer/receiver/presenter/AccountInfoPresenter.java [hunk 6]
```diff
@@ -164,19 +164,19 @@ private void checkTaskState() {
         @Override
         protected TaskResponse doInBackground(Void... args) {
             TaskResponse response = new TaskResponse();
-            String address = null;
+            String info = null;
 
             if (accountInfo != null) {
-                address = accountInfo.getPublicAddress();
+                info = accountInfo.getAccountInfo();
```
Comment (shayb) [inline transfer/src/main/java/org/kinecosystem/transfer/receiver/presenter/AccountInfoPresenter.java]: i wish we had better naming
accountInfo.getAccountInfo();

getData() ?

**Pull request merged**